pkg/api/types.go

```diff
@@ -1345,6 +1345,16 @@ type NodeSpec struct {
 	Unschedulable bool `json:"unschedulable,omitempty"`
 }
 
+// KubeletPorts lists ports opened by Kubelet
+type KubeletPorts struct {
+	// KubeletPort on which Kubelet is listening
+	KubeletPort uint `json:kubeletPort`
+	// KubeletReadOnlyPorty only port open on Kubelet
+	KubeletReadOnlyPorty uint `json:kubeletReadOnlyPort`
```

Comment:
(If it wasn't clear, I'd really hate for this to become part of the stable v1 API so that we can't remove it)